packages/data-context/src/actions/ElectronActions.ts

```diff
@@ -4,6 +4,13 @@ import type { DataContext } from '..'
 import _ from 'lodash'
 import path from 'path'
 import assert from 'assert'
+import debugLib from 'debug'
+
+const debug = debugLib('cypress:data-context:ElectronActions')
+
+// Declare notification variable globally so that it doesn't get garbage collected
+// before the user clicks on the notification
+let notification
```

Comment:
I had the same thought, but using a `Map` instead of an array. We could give each one a uuid and do `map.set(uuid, notification). When the click handler is called, we could just do `map.delete(uuid)`.

I don't think we ever want to clear them as long as the App is open, since at least in MacOS, you can stack notifications forever:

![image](https://github.com/cypress-io/cypress/assets/19196536/3ab402f3-7e76-4601-8908-c2ae833c231b)

Perhaps someone comes back after the weekend and wants to see what failed when they pushed code and clocked off on Friday afternoon.

Reply:
The map is much better!

Reply:
Actually, we can do better with `Set`. Those are unique by reference. We can just do `set.add(notification)`, and when you click it, `set.delete(notification)`.